### PR TITLE
[t4] add source table name to foreign key identifier

### LINKED ISSUE #2752: key.t.table.ForeignKeys.Add
Describe your issue.

https://github.com/linq2db/linq2db/blob/master/Source/LinqToDB.Templates/DataModel.ttinclude
line 247 249 
		key.t.table.ForeignKeys.Add(
			key.k.OtherTable.IsDefaultSchema ? key.KeyName : key.k.OtherTable.SchemaName + "." + key.KeyName,
			key.key);

### Steps to reproduce
Postgress database table having same  foreign key name throwing error/failing at above line

### Environment details
linq2db version: 3.2.2
Database Server: Postgress 11.6
Database Provider: Postgress
Operating system: windows
.NET Framework: 3.1

### PR DESCRIPTION
**Pull request by MaceWindu**
Fix #2752

### CODE REVIEW

File: Tests/Tests.T4/Databases/PostgreSQL.11.generated.cs [hunk 1]
```diff
@@ -138,62 +141,8 @@ public partial class TestTableFunction1Result
...
-			public int[]?                      intarray            { get; set; }
-			public int[]?                      int2darray          { get; set; }
-			public long[]?                     longarray           { get; set; }
-			public TimeSpan[]?                 intervalarray       { get; set; }
-			public double[]?                   doublearray         { get; set; }
-			public decimal[]?                  numericarray        { get; set; }
-			public decimal[]?                  decimalarray        { get; set; }
-		}
+// Use 'GenerateProcedureErrors=false' to disable errors.
+#error Couldn't find PostgreSQL type with OID 1412796
```
Comment (MaceWindu) [inline Tests/Tests.T4/Databases/PostgreSQL.11.generated.cs:145]: This is npgsql error (looks like regression in npgsql)

Comment (MaceWindu) [reply Tests/Tests.T4/Databases/PostgreSQL.11.generated.cs]: actually it was due missing pre-load of  custom  types by npgsql

**Review by viceroypenguin — APPROVED** [review]

**Review by sdanyliv — APPROVED** [review]

**Pull request merged**